dashboards-observability/public/components/event_analytics/explorer/explorer.tsx

```diff
@@ -1279,6 +1285,17 @@ export const Explorer = ({
     [tempQuery]
   );
 
+  const generateViewQuery = (query: string) => {
+    if (query.includes(appBaseQuery)) {
+      if (query.includes('|')) {
+        // Some scenarios have ' | ' after base query and some have '| '
+        return query.replace(' | ', '| ').replace(appBaseQuery + '| ', '');
```

Comment:
is the if necessary? also this seems flakey. it doesn't handle two spaces before `|`, or how does it ensure there's only 0 or 1 space before `|`?

Reply:
All query are pipelines, no matter how they're composed or split.

Can we instead try to use set-wise operations here ; 

"...pipeline string...".split('|').map {|token| token.trim() }

And then later

tokens.join("|") 


???

Reply:
It is also quite possible that the query can have multiple pipes, `source = opensearch_dashboards_sample_logs | where response='403' | sort machine`. In this case the replace would only replace the first pipe and therefore the second replace will not match anything and the base query will not be removed from the search bar view.

Reply:
@pjfitzgibbons not sure where is the "set", but how do you handle ``source=index | fields `a|b` ``?
i brought up a similar issue before: https://github.com/opensearch-project/observability/pull/81#discussion_r680277806

Reply:
Hi @eugenesk24 ,

As per the conversations on this PR we have made an implementation which would work irrespective of any amount of spaces present before or after the pipe. But if any field exists with a pipe sign in it like 'a |b' or 'a| b' this will fail. We wanted to confirm if these could possibly be valid fields or not.
Implementation:
  const generateViewQuery = (query: string) => {
    if (query.includes(appBaseQuery)) {
      if (query.includes('|')) {
        const newQuery = query
          .split('|')
          .map((section: string) => section.trim())
          .join('|');
        return newQuery.replace(appBaseQuery + '|', '');
      }
      return '';
    }
    return query;
  };